### DISCUSSION TITLE
Marker Customization

### DISCUSSION THREAD

**Discussion by bundleberk** [patrykandpatrick/vico#584] [Questions]
I want to add another text below the y value that is shown in the Marker. How can I achieve that?

**Comment by patrickmichalik** [comment]
Hello! Just to confirm, are you using `MarkerComponent` for the `Marker`?

**Reply by bundleberk** [reply]
Hi, I am using MarkerComponent and I want to customise the marker's view as a whole if possible, not just its text. I am using jetpack compose by the way.

**Reply by patrickmichalik** [reply]
Got it. What approach should be taken depends on what changes are needed. You can add an extra line of text via `MarkerLabelFormatter`. `getLabel` returns a `CharSequence`, so you can use `SpannableString` to give this line a different style. If the desired result can’t be achieved by these means, you can create a custom `Marker` implementation, possibly based on `MarkerComponent`. In Vico 2, creating a `MarkerComponent` subclass is also a viable option—all private members have been made protected.

It’s a similar story for the rest of the `Marker`. `MarkerComponent` lets you pass in a custom `Component` for each of its parts, giving you quite a bit of flexibility. If you need an entirely different layout, however, a custom `Marker` implementation or a `MarkerComponent` subclass may be needed.

If you share what changes you’re looking to make, I’ll be happy to suggest an approach.

**Reply by joseandroidengineer** [reply]
Hello! Is it possible to show the x-value next to or underneath the y-value text in the same marker? I'm using the DefaultCartesianMarker to show the x-value. 

Thank you for your help! 😄

**Reply by Gowsky** [reply]
@joseandroidengineer yes, by using `CartesianMarkerValueFormatter` and `CartesianMarker.Target.x`.

**Reply by joseandroidengineer** [reply]
Just the answer I was looking for. Thanks!
